src/benchmark/file_io_read_mmap_benchmark.cpp

```diff
@@ -137,10 +139,79 @@ BENCHMARK_DEFINE_F(FileIOMicroReadBenchmarkFixture, MMAP_ATOMIC_MAP_SHARED_SEQUE
   close(fd);
 }
 
+BENCHMARK_DEFINE_F(FileIOMicroReadBenchmarkFixture, UMAP_ATOMIC_MAP_PRIVATE_RANDOM)(benchmark::State& state) {
+  auto fd = int32_t{};
+  Assert(((fd = open(filename, O_RDONLY)) >= 0), fail_and_close_file(fd, "Open error: ", errno));
+
+  for (auto _ : state) {
+    state.PauseTiming();
+    micro_benchmark_clear_disk_cache();
+    state.ResumeTiming();
+
+    // Getting the mapping to memory.
+    const auto OFFSET = off_t{0};
+
+    auto* map = reinterpret_cast<int32_t*>(umap(NULL, NUMBER_OF_BYTES, PROT_READ, MAP_PRIVATE, fd, OFFSET));
+    Assert((map != MAP_FAILED), fail_and_close_file(fd, "Mapping failed: ", errno));
+
+    // madvise(map, NUMBER_OF_BYTES, MADV_SEQUENTIAL);
+
+    auto sum = uint64_t{0};
+    for (size_t index = 0; index < NUMBER_OF_ELEMENTS; ++index) {
+      sum += map[index];
+    }
+
+    state.PauseTiming();
+    Assert(control_sum == sum, "Sanity check failed: Not the same result");
+    state.ResumeTiming();
+
+    // Remove memory mapping after job is done.
+    Assert((uunmap(map, NUMBER_OF_BYTES) == 0), fail_and_close_file(fd, "Unmapping failed: ", errno));
+  }
+
+  close(fd);
+}
+
+BENCHMARK_DEFINE_F(FileIOMicroReadBenchmarkFixture, UMAP_ATOMIC_MAP_PRIVATE_SEQUENTIAL)(benchmark::State& state) {
+  auto fd = int32_t{};
+  Assert(((fd = open(filename, O_RDONLY)) >= 0), fail_and_close_file(fd, "Open error: ", errno));
+
+  for (auto _ : state) {
+    state.PauseTiming();
+    micro_benchmark_clear_disk_cache();
+    state.ResumeTiming();
+
+    // Getting the mapping to memory.
+    const auto OFFSET = off_t{0};
+
+    auto* map = reinterpret_cast<int32_t*>(umap(NULL, NUMBER_OF_BYTES, PROT_READ, MAP_PRIVATE, fd, OFFSET));
+    Assert((map != MAP_FAILED), fail_and_close_file(fd, "Mapping failed: ", errno));
+
+    // madvise(map, NUMBER_OF_BYTES, MADV_SEQUENTIAL);
+
+    auto sum = uint64_t{0};
+    for (size_t index = 0; index < NUMBER_OF_ELEMENTS; ++index) {
+      sum += map[index];
+    }
+
+    state.PauseTiming();
+    Assert(control_sum == sum, "Sanity check failed: Not the same result");
+    state.ResumeTiming();
+
+    // Remove memory mapping after job is done.
+    Assert((uunmap(map, NUMBER_OF_BYTES) == 0), fail_and_close_file(fd, "Unmapping failed: ", errno));
+  }
+
+  close(fd);
+}
```

Comment:
Maybe you could refactor this so that it matches the structure in the newly refactored changes? As it seems, we only use umap as a call differently to the previous mmap calls, so perhaps pass a flag if umap is supposed to be used? This would reduce code lines and ease further developments.